internal/uvm/update_uvm.go

```diff
@@ -0,0 +1,57 @@
+package uvm
+
+import (
+	"context"
+
+	hcsschema "github.com/Microsoft/hcsshim/internal/schema2"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+)
+
+func (uvm *UtilityVM) UpdateConstraints(ctx context.Context, data interface{}, annotations map[string]string) error {
+	var memoryLimitInBytes *uint64
+	var processorLimits *hcsschema.ProcessorLimits
+
+	switch resources := data.(type) {
+	case *specs.WindowsResources:
+		if resources.Memory != nil {
+			memoryLimitInBytes = resources.Memory.Limit
+		}
+		if resources.CPU != nil {
+			processorLimits := &hcsschema.ProcessorLimits{}
+			if resources.CPU.Maximum != nil {
+				processorLimits.Limit = uint64(*resources.CPU.Maximum)
+			}
+			if resources.CPU.Shares != nil {
+				processorLimits.Weight = uint64(*resources.CPU.Shares)
+			}
+		}
+	case *specs.LinuxResources:
+		if resources.Memory != nil {
+			mem := uint64(*resources.Memory.Limit)
+			memoryLimitInBytes = &mem
+		}
+		if resources.CPU != nil {
+			processorLimits := &hcsschema.ProcessorLimits{}
+			if resources.CPU.Quota != nil {
+				processorLimits.Limit = uint64(*resources.CPU.Quota)
+			}
+			if resources.CPU.Shares != nil {
+				processorLimits.Weight = uint64(*resources.CPU.Shares)
+			}
+		}
+	}
+
+	if memoryLimitInBytes != nil {
+		memoryLimit := uvm.normalizeMemorySize(ctx, *memoryLimitInBytes)
```

Comment:
`UpdateMemory` call below also does the memory size normalization. Do we still need to do that here?

Reply:
:o good point. Must have been a left over from earlier iterations.